ramp-engine/ramp_engine/conda.py

```diff
@@ -0,0 +1,68 @@
+# All the following functions may be used with dask, so
+# they should include imports inside functions.
+from typing import Dict
+
+
+def _conda_info_envs() -> Dict:
```

Comment:
I am not using `virtualenv` but do you think that it would be possible to have a higher-level API that could unify either using conda or virtualenv (of course not in this PR and anyway all these stuff are private).

Reply:
yeah, there is nothing too conda specific in workers. We just need to compute the path to the python executable inside the virtualenv. The only issue is that virtualenv doesn't have a CLI to list all available envs. So we would probably need to use some additional tool on top, maybe [poetry](https://python-poetry.org/docs/managing-environments/#listing-the-environments-associated-with-the-project) or [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/command_ref.html?highlight=lsvirtualenv#lsvirtualenv) (unless we are OK passing the path to the virtualenv in the config).